frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx

```diff
@@ -0,0 +1,210 @@
+      expect(screen.queryByText("is")).not.toBeInTheDocument();
+    });
+
+    it("should allow to remove a filter", () => {
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+      });
+
+      userEvent.click(screen.getByRole("checkbox", { name: "True" }));
+      userEvent.click(screen.getByRole("checkbox", { name: "True" }));
+      expect(screen.getByRole("checkbox", { name: "True" })).not.toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "False" })).not.toBeChecked();
+      expect(getNextFilterName()).toBeNull();
+
+      userEvent.click(screen.getByRole("checkbox", { name: "False" }));
+      userEvent.click(screen.getByRole("checkbox", { name: "False" }));
+      expect(screen.getByRole("checkbox", { name: "True" })).not.toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "False" })).not.toBeChecked();
+      expect(getNextFilterName()).toBeNull();
+    });
+  });
+
+  describe("existing filter", () => {
+    it("should update a filter with one value", () => {
+      const { query, stageIndex, column, filter } = createQueryWithFilter({
+        operator: "=",
+        values: [true],
+      });
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+        filter,
+      });
+      expect(screen.getByRole("checkbox", { name: "True" })).toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "False" })).not.toBeChecked();
+
+      userEvent.click(screen.getByRole("checkbox", { name: "False" }));
+      expect(getNextFilterName()).toBe("Is trial is false");
+      expect(screen.getByRole("checkbox", { name: "True" })).not.toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "False" })).toBeChecked();
+    });
+
+    it("should update a filter with no values", async () => {
+      const { query, stageIndex, column, filter } = createQueryWithFilter({
+        operator: "is-null",
+        values: [],
+      });
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+        filter,
+      });
+      expect(screen.getByText("is empty")).toBeInTheDocument();
+      expect(screen.getByRole("checkbox", { name: "True" })).not.toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "False" })).not.toBeChecked();
+
+      userEvent.click(screen.getByText("is empty"));
+      userEvent.click(await screen.findByText("Not empty"));
+      expect(getNextFilterName()).toBe("Is trial is not empty");
+      expect(screen.getByText("not empty")).toBeInTheDocument();
+      expect(
+        await screen.findByRole("checkbox", { name: "True" }),
+      ).not.toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "False" })).not.toBeChecked();
+
+      userEvent.click(screen.getByText("not empty"));
+      userEvent.click(await screen.findByText("Is"));
+      userEvent.click(await screen.findByRole("checkbox", { name: "True" }));
```

Comment:
I remember we discussed `findByRole` that it's slow, can't you use `byLabelText`?

Reply:
Not unless `aria-label` is set explicitly. It seems there is no other way to do it without adding it.